app/attributes/type/value_object.rb

```diff
@@ -0,0 +1,42 @@
+module Type
+  class ValueObject < ActiveModel::Type::Value
+    attr_reader :source
+
+    def initialize(*args, **kwargs)
+      @source = kwargs.delete(:source)
+      super
+    end
+
+    def type
+      :value_object
+    end
+
+    def serialize(value)
+      value.to_s
+    end
+
+    def type_cast_for_schema(value)
```

Comment:
Looking at the docs this is a `private` method that has been overridden here, and the docs say that they were trying to remove it - wondering if there is an alternative that might have more longevity.

https://apidock.com/rails/ActiveRecord/Type/Value/type_cast_for_schema

Reply:
Yep I read that. But it is not private, the comment is misleading, if you take a look at the class source code the method is public 🤣 I was overriding out of excessive precaution mostly, as the default method from the `ActiveModel::Type::Value` superclass does an `inspect` on the value, but that's probably not what we want on a ValueObject.

I'm inclined to remove the method completely. I don't think is being used and also I'm not clear what exactly "cast a value for schema dumping" means. In all my tests and debugging sessions that method is never called. Maybe is called under certain scenarios that I can't foresee now but we can always get back to this an change behaviour 🤔

Reply:
Curious how you caught this - how did you know to think to overwrite the method?

Reply:
Yeah I'm wondering if it might be more confusing to leave it in. But you never know it might stop the app from blowing up for obscure reason so I have no strong opinion either way really